sloghandler.go

```diff
@@ -127,21 +123,50 @@ func (l *slogHandler) WithGroup(name string) slog.Handler {
 	if l.sink == nil {
 		return l
 	}
+	if name == "" {
+		// slog says to inline empty groups
+		return l
+	}
 	clone := *l
 	if l.slogSink != nil {
 		clone.slogSink = l.slogSink.WithGroup(name)
 		clone.sink = clone.slogSink
 	} else {
-		clone.groupPrefix = clone.addGroupPrefix(name)
+		clone.groupPrefix = addPrefix(clone.groupPrefix, name)
 	}
 	return &clone
 }
 
-func (l *slogHandler) addGroupPrefix(name string) string {
-	if l.groupPrefix == "" {
+// attrToKVs appends a slog.Attr to a logr-style kvList.  It handle slog Groups
+// and other details of slog.
+func attrToKVs(attr slog.Attr, groupPrefix string, kvList []any) []any {
+	attrVal := attr.Value.Resolve()
+	if attrVal.Kind() == slog.KindGroup {
```

Comment:
I suppose this is the "retools sloghandler to handle slog Attrs a bit better" part.

Looks okay to me now. Let's rebase and then merge it without further refactoring (your plan B).